### PR TITLE
Added Application and Application Owner as metadata to the Consumer r…

### PR DESCRIPTION
**Pull request by Elson9**
…ecord

---
🚀 Feature branch deployment: https://api-services-portal-feature-app-metadata.apps.silver.devops.gov.bc.ca

### CODE REVIEW

**Comment by Elson9** [comment]
Fixes #980

File: src/services/kong/consumer-service.ts [hunk 1]
```diff
@@ -51,16 +52,20 @@ export class KongConsumerService {
       return { created: false, consumer: result };
     } catch (err) {
       logger.debug('createOrGetConsumer - CATCH ERROR %s', err);
-      const result = await this.createKongConsumer(username, customId);
+      const result = await this.createKongConsumer(username, customId, app);
       logger.debug('createOrGetConsumer - CATCH RESULT %j', result);
       return { created: false, consumer: result };
```
Comment (Elson9) [inline src/services/kong/consumer-service.ts:57]: Should `created: true` here?

Comment (ikethecoder) [reply src/services/kong/consumer-service.ts:57]: yes think so

File: src/services/kong/consumer-service.ts [hunk 2]
```diff
@@ -51,16 +52,20 @@ export class KongConsumerService {
...
     }
   }
 
-  public async createKongConsumer(username: string, customId: string) {
+  public async createKongConsumer(
+    username: string,
+    customId: string,
+    app: any
```
Comment (ikethecoder) [inline src/services/kong/consumer-service.ts:64]: Instead of "any", should be able to use "Application"

Comment (Elson9) [reply src/services/kong/consumer-service.ts]: Fixed

File: src/services/kong/consumer-service.ts [hunk 3]
```diff
@@ -51,16 +52,20 @@ export class KongConsumerService {
...
+  ) {
     let body: KongConsumer = {
       username: username,
-      tags: ['aps-portal'],
+      tags: ['aps-portal', `app:${app.name}`, `owner:${app.owner.name}`],
```
[the remark this answers is not in the document]
Comment (Elson9) [reply src/services/kong/consumer-service.ts]: Since owner is part of the app object, wouldn't it only exist if app isn't null?

**Review by rustyjux — APPROVED** [review]

**Review by ikethecoder — APPROVED** [review]

**Pull request merged**